lib/puppet.rb

```diff
@@ -207,12 +207,39 @@ def self.base_context(settings)
         require 'puppet/network/http'
         Puppet::Network::HTTP::NoCachePool.new
       },
+      :ssl_context => proc {
+        begin
+          build_ssl_context
+        rescue => e
+          Puppet.log_exception(e, "Failed to initialize SSL: #{e.message}")
+          Puppet.err("Run `puppet agent -t`")
+          raise e
+        end
+      },
       :ssl_host => proc { Puppet::SSL::Host.localhost },
       :plugins => proc { Puppet::Plugins::Configuration.load_plugins },
       :rich_data => false
     }
   end
 
+  def self.build_ssl_context
+    cert = Puppet::X509::CertProvider.new
+    cacerts = cert.load_cacerts
+    raise Puppet::Error, "The CA certificate is missing from '#{Puppet[:localcacert]}'" unless cacerts
```

Comment:
Should these error messages be translated?

Reply:
yeah, I realized I missed some earlier. I can add a commit for this